tests/integration_test.rs

```diff
@@ -0,0 +1,30 @@
+use assert_cmd::Command;
+use tempdir::TempDir;
+
+#[test]
+fn test_end2end_execution() {
+    let mut cmd = Command::cargo_bin("mongo-task-generator").unwrap();
+    let tmp_dir = TempDir::new("generated_resmoke_config").unwrap();
+
+    cmd.args(&[
+        "--target-directory",
+        tmp_dir.path().to_str().unwrap(),
+        "--expansion-file",
+        "tests/data/sample_expansions.yml",
+        "--evg-project-file",
+        "tests/data/evergreen.yml",
+        "--evg-auth-file",
+        "tests/data/sample_evergreen_auth.yml",
+        "--resmoke-command",
+        "python3 tests/mocks/resmoke.py",
+        "--use-task-split-fallback",
+    ])
+    .assert()
+    .success();
+
+    let tmp_dir_path = tmp_dir.path();
+    assert!(tmp_dir_path.exists());
+
+    let files = std::fs::read_dir(tmp_dir_path).unwrap();
+    assert_eq!(1111, files.into_iter().collect::<Vec<_>>().len());
```

Comment:
I'm not quite understand this line. Is this assert trying to verify total amount of files are not 1111? If so what is 1111 represent. If not, what is it for?

Reply:
I'm wondering do we know there are exactly 1111 files being generated? Base on the number of test files under /tests/data/tests?

Reply:
I ran the generator against the sample data to tell. 

We could calculate it, but the calculation gets complicated especially since we are running against a real evergreen.yml. But basically each remoke task will generate 6 files unless they are marked as multiversion, in which case they will generate 6 * (the number of multiversion fixtures configured). The multiversion value can be different depending on the task type. Also, there will be 1 file for the overall configuration.